widgetssdk/src/main/java/com/glia/widgets/call/CallController.java

```diff
@@ -231,7 +231,9 @@ public void onHoldChanged(boolean isOnHold) {
     public void engagementEnded() {
         Logger.d(TAG, "engagementEndedByOperator");
         stop();
-        dialogController.dismissDialogs();
+        if (!Dependencies.getUseCaseFactory().createIsOngoingEngagementUseCase().invoke()) {
```

Comment:
`IsOngoingEngagementUseCase` is present in both controllers, you can just call `if(!isOngoingEngagementUseCase())`